c3d/src/C3d.cpp

```diff
@@ -544,21 +659,43 @@ class C3dVisitor : public RecursiveASTVisitor<C3dVisitor> {
     // Printing parameters
     enum { BeforeLParen, InArgs } State = BeforeLParen;
     for (const auto& A: Parameters) {
+      const int shift = strlen("c3d_parameter:");
       switch (State) {
         case BeforeLParen:
           Out << " (";
-          Out << A->getAnnotation().slice(14, A->getAnnotation().size());
+          Out << A->getAnnotation().slice(shift, A->getAnnotation().size());
           State = InArgs;
           break;
         case InArgs:
           Out << ", ";
-          Out << A->getAnnotation().slice(14, A->getAnnotation().size());
+          Out << A->getAnnotation().slice(shift, A->getAnnotation().size());
           break;
       }
     }
     if (State == InArgs)
       Out << ")";
 
+    // Printing where clause, joining them into a single conjunction
+    // since that's what 3d expects.
+    if (! WhereClauses.empty()) {
+        enum { First, Mid } State = First;
+        const int shift = strlen("c3d_where:");
+        Out << "\nwhere ";
+        for (const auto& W: WhereClauses) {
+            switch (State) {
+                case Mid:
+                    Out << " && ";
+                    // fall through!
+
+                case First:
+                    Out << "(" << W->getAnnotation().slice(shift, W->getAnnotation().size()) << ")";
```

Comment:
this adds superfluous parentheses when there's only a single where clause? is this desirable? (quite possibly)

Reply:
I wasn't super sure if it's OK to not wrap them. I think it is, since the thing inside must be a plain expression at this point, but I'd like to double check with the grammars before removing this.